src/utils/utils_math.cpp

```diff
@@ -0,0 +1,24 @@
+/*
```

Comment:
Is there a reason for this file to be in C++? I think we should default to C if possible - we might have use cases where we don't want any dependency on C++ (runtime or compiler)

Reply:
Done.